src/hotspot/cpu/x86/x86.ad

```diff
@@ -4158,8 +4158,8 @@ instruct scatter_masked(memory mem, vec src, vec idx, kReg mask, kReg ktmp, rReg
 
 // Replicate byte scalar to be vector
 instruct vReplB_reg(vec dst, rRegI src) %{
-  predicate(UseAVX >= 2);
-  match(Set dst (ReplicateB src));
+  predicate(UseAVX >= 2 && Matcher::vector_element_basic_type(n) == T_BYTE);
+  match(Set dst (Replicate src));
```

Comment:
Integral gather patterns can be merged by folding predicates into macro assembler.

Reply:
@jatin-bhateja can you spell out a bit what you mean?
And when you mention broadcast instructions, do you mean any non Replicate instructions?

Reply:
@eme64, I see an opportunity for merging some integral broadcast patterns which share the operand definitions by moving predicates into macro assembler layer.

Reply:
https://github.com/openjdk/jdk/pull/14830#discussion_r1371481240
@e1iu have you addressed this? What do you think?

Reply:
@jatin-bhateja @eme64 I think it's very x86 specifically. I don't have too much confidence changing other platforms' code. So as I mention at beginning, this patch only adds predicate on each rules to pursue lower risks. I think This's a workable way to refactor some shared code.

Reply:
@jatin-bhateja would it be ok if you did this as a follow-up RFE? Or otherwise could you give very clear instructions about what exactly you want to see, @jatin-bhateja ?

Reply:
[replicate_cleanup_x86..txt](https://github.com/openjdk/jdk/files/13264475/replicate_cleanup_x86.txt)
@e1iu , I meant something is these lines, at the minimal combine the byte / short patterns having same operands.

Reply:
@jatin-bhateja I understand now. Would it be ok if we did this in a separate RFE? Then we can limit this to x86.